src/Arcus.EventGrid.Tests.Integration/Publishing/EventPublishingTests.cs

```diff
@@ -191,6 +220,78 @@ await EventGridPublisherBuilder
                 });
         }
 
+        [Fact]
+        public async Task PublishMultipleRawEvents_WithBuilder_ValidParameters_SucceedsWithRetryCount()
+        {
+            // Arrange
+            var topicEndpoint = Configuration.GetValue<string>("Arcus:EventGrid:TopicEndpoint");
+            var endpointKey = Configuration.GetValue<string>("Arcus:EventGrid:EndpointKey");
+            var events =
+                Enumerable
+                    .Repeat<Func<Guid>>(Guid.NewGuid, 2)
+                    .Select(newGuid => new RawEvent(
+                                newGuid().ToString(),
+                                subject: "integration-test",
+                                body: "{\"licensePlate\": \"1-TOM-1337\"}",
+                                type: "Arcus.Samples.Cars.NewCarRegistered",
+                                dataVersion:"1.0",
+                                eventTime: DateTimeOffset.Now))
+                    .ToArray();
+
+            // Act
+            await EventGridPublisherBuilder
+                  .ForTopic(topicEndpoint)
+                  .UsingAuthenticationKey(endpointKey)
+                  .Build()
+                  .PublishManyAsync(events);
+
+            // Assert
+            Assert.All(
```

Comment:
Same here, prefer not to do inline

Reply:
You mean inside the `Assert.All`? or the re-deserialize from the event body?

Reply:
This should be moved to a dedicated method imo as it decreases readability:
```
rawEvent =>
{
    TracePublishedEvent(rawEvent.Id, events);
    string receivedEvent = _serviceBusEventConsumerHost.GetReceivedEvent(rawEvent.Id, 5);
    AssertReceivedEvent(rawEvent.Id, rawEvent.EventType, rawEvent.Subject, licensePlate, receivedEvent);
}
```

Reply:
See that other tests do same, let's align them